apps/api/src/billing/services/trial-validation/trial-validation.service.ts

```diff
@@ -1,30 +1,25 @@
+import { GroupSpec } from "@akashnetwork/akash-api/akash/deployment/v1beta3";
 import * as v1beta3 from "@akashnetwork/akash-api/v1beta3";
 import { EncodeObject } from "@cosmjs/proto-signing";
 import assert from "http-assert";
 import { singleton } from "tsyringe";
 
 import { UserWalletOutput } from "@src/billing/repositories";
-import { AUDITOR, TRIAL_ATTRIBUTE } from "@src/deployment/config/provider.config";
+import { AUDITOR, TRIAL_ATTRIBUTE, TRIAL_REGISTERED_ATTRIBUTE } from "@src/deployment/config/provider.config";
 import { getAddressDeployments } from "@src/services/external/apiNodeService";
 
 const TRIAL_DEPLOYMENT_LIMIT = 5;
 
 @singleton()
 export class TrialValidationService {
   async validateLeaseProviders(decoded: EncodeObject, userWallet: UserWalletOutput) {
-    if (userWallet.isTrialing && decoded.typeUrl === "/akash.deployment.v1beta3.MsgCreateDeployment") {
+    if (decoded.typeUrl === "/akash.deployment.v1beta3.MsgCreateDeployment") {
       const value = decoded.value as v1beta3.MsgCreateDeployment;
-
-      value.groups.forEach(group => {
-        const hasTrial = group.requirements.attributes.some(attribute => {
-          return attribute.key === TRIAL_ATTRIBUTE && attribute.value === "true";
-        });
-
-        const hasSignedByAllOf = group.requirements.signedBy.allOf.every(signedBy => {
-          return signedBy === AUDITOR;
-        });
-        assert(hasTrial && hasSignedByAllOf, 400, `provider not authorized: ${group.requirements.attributes}`);
-      });
+      if (userWallet.isTrialing) {
+        this.validateAttribute(value.groups, TRIAL_ATTRIBUTE);
+      } else if (!userWallet.stripeCustomerId) {
```

Comment:
why is this check needed? `isTrialing` should be enough. it's put to `false` once user does a first top up, meaning there is a stripe customer at this point.

Reply:
```suggestion
      } else {
```